lib/dockerregistry/inventory.go

```diff
@@ -872,6 +880,63 @@ func getRegistryTagsFrom(req stream.ExternalRequest) (*google.Tags, error) {
 	return tags, nil
 }
 
+func getGManifestListWrapper(
+	req stream.ExternalRequest) (*GManifestList, error) {
+
+	var gManifestList *GManifestList
+
+	var getGManifestListCondition wait.ConditionFunc = func() (bool, error) {
+		var err error
+
+		gManifestList, err = getGManifestListFrom(req)
+
+		// We never return an error (err) in the second part of our return
+		// argument, because we don't want to prematurely stop the
+		// ExponentialBackoff() loop; we want it to continue looping until
+		// either we get a well-formed value, or until it hits
+		// ErrWaitTimeout. This is how ExponentialBackoff() uses the
+		// ConditionFunc type.
+		if err == nil &&
+			gManifestList != nil &&
+			len(gManifestList.Manifests) > 0 {
```

Comment:
Aside: is it possible to have a manifest list with no manifests?  Maybe this is what you're hitting on the odd real-world cases.

Reply:
Good point. I wrote down your thoughts in my debugging notebook!